Source/Core/InputCommon/ControlReference/ExpressionParser.cpp

```diff
@@ -92,12 +92,16 @@ Lexer::Lexer(std::string expr_) : expr(std::move(expr_))
   it = expr.begin();
 }
 
-std::string Lexer::FetchDelimString(char delim)
+Token Lexer::GetDelimitedToken(TokenType type, char delimeter)
 {
-  const std::string result = FetchCharsWhile([delim](char c) { return c != delim; });
-  if (it != expr.end())
-    ++it;
-  return result;
+  std::string value;
+  value += FetchCharsWhile([&](char c) { return c != delimeter && c != '\n'; });
```

Comment:
Is there a reason you're `+=`ing onto the string instead of just constructing it from the return value?

Reply:
Nope. I'm not really sure how I came up with that. :) I'll fix it.